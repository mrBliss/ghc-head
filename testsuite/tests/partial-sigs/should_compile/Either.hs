{-# LANGUAGE PartialTypeSignatures #-}
module Either where

barry :: _a -> (_b _a, _b _)
barry x = (Left "x", Right x)

test :: ((Either [Char] Bool, Either [Char] Bool), (Either [Char] Char, Either [Char] Char))
test = (barry True, barry 'c')
