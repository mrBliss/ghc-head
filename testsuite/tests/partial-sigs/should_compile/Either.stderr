TYPE SIGNATURES
  barry ::
    forall w_a. w_a -> (Either [Char] w_a, Either [Char] w_a)
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
